qa/L0_server_status/server_status_test.py

```diff
@@ -111,17 +118,30 @@ def test_model_latest_infer(self):
 
             # Initially there should be no version stats..
             try:
-                for pair in [("localhost:8000", ProtocolType.HTTP), ("localhost:8001", ProtocolType.GRPC)]:
-                    server_status0, req_id0 = _get_server_status(pair[0], pair[1], model_name)
-                    self.assertTrue(model_name in server_status0.model_status,
-                                    "expected status for model " + model_name)
-                    self.assertEqual(len(server_status0.model_status[model_name].version_status), 3,
-                                    "expected status for 3 versions for model " + model_name)
+                for pair in [("localhost:8000", "http"), ("localhost:8001", "grpc")]:
+                    if pair[1] == "http":
+                        triton_client = httpclient.InferenceServerClient(
+                            url=pair[0], verbose=True)
+                    else:
+                        triton_client = grpcclient.InferenceServerClient(
+                            url=pair[0], verbose=True)
+
+                    self.assertTrue(triton_client.is_server_live())
+                    self.assertTrue(triton_client.is_server_ready())
+                    model_metadata = triton_client.get_model_metadata(model_name)
```

Comment:
Since you didn't specify version here you should be the metadata for the latest version, correct? You should add some assertions to check that.

Reply:
@deadeyegoodwin The metadata reports all the versions in this case. Should it not be reporting all the versions?

Reply:
The spec here is a little strange. For triton our metadata is always the same no matter the version requested. So I would say that the version that is in the request we should just use to see if the entire request should pass/fail (that is, if version is given and that version isn't actually available then fail). But otherwise the actual content returned for the request should always be the same, no matter if a version if requested or not.

Reply:
To clarify -> This would mean that the content for each valid version/if no version is specified should contain all the versions that the model supports (and not just the specified/latest version)?

Reply:
The versions listed in the metadata response are just a list of the version numbers that are available for that model. It depends on the version policy of the model. I'm saying that the requested version (if any) does not change what we return in that version list. The requesed version (if any) should just be used to fail the request if that version is not valid for the model. So, in practice, with Triton it doesn't make much sense to use a version in the metadata call (but we should still test it).